lib/src/cupertino/cupertino_popover_menu.dart

```diff
@@ -349,21 +368,49 @@ class RenderPopover extends RenderShiftedBox {
     );
   }
 
+  @override
+  void dispose() {
+    _clipPathLayer.layer = null;
+    super.dispose();
+  }
+
+  Path _buildBorderPathForOffset(Offset offset) {
+    late ArrowDirection direction;
+    late double arrowCenter;
+
+    final localFocalPoint = focalPoint.globalOffset != null ? globalToLocal(focalPoint.globalOffset!) : null;
+    if (localFocalPoint != null) {
+      // We have a menu focal point. Orient the arrow towards that
+      // focal point.
+      direction = _computeArrowDirection(Offset.zero & size, localFocalPoint);
+      arrowCenter = _computeArrowCenter(direction, localFocalPoint);
+    } else {
+      // We don't have a menu focal point. Perhaps this is a moment just
+      // before, or just after a focal point becomes available. Until then,
+      // render with the arrow pointing down from the center of the toolbar,
+      // as an arbitrary arrow position.
+      direction = ArrowDirection.down;
+      arrowCenter = 0.5;
+    }
+
+    return _buildBorderPath(direction, arrowCenter);
```

Comment:
Is there any particular reason that all of this was moved down here to its own method?